### PR TITLE
Miscellaneous CNV bug fixes and cleanup.

### LINKED ISSUE #645: Change BED input to AnnotateTargets.

### LINKED ISSUE #684: AllelicCNV should throw a proper exception when the number of Hets is low
When running the WGS ACNV workflow on the highly downsampled samples, and using GetBayesianHetCoverage, which returned fewer hets(only 7) than GetHetCoverage, the AllelicCNV produces the following: 

```
org.apache.commons.math3.exception.NotStrictlyPositiveException: 0 is smaller than, or equal to, the minimum (0)
at org.apache.commons.math3.linear.AbstractRealMatrix.<init>(AbstractRealMatrix.java:68)
at org.apache.commons.math3.linear.Array2DRowRealMatrix.<init>(Array2DRowRealMatrix.java:60)
at org.broadinstitute.hellbender.tools.exome.SNPSegmenter.writeSegmentFile(SNPSegmenter.java:54)
at org.broadinstitute.hellbender.tools.exome.AllelicCNV.calculateAndWriteSNPSegmentation(AllelicCNV.java:370)
at org.broadinstitute.hellbender.tools.exome.AllelicCNV.performSNPSegmentationStep(AllelicCNV.java:340)
at org.broadinstitute.hellbender.tools.exome.AllelicCNV.runPipeline(AllelicCNV.java:263)
at org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram.doWork(SparkCommandLineProgram.java:38)
at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:108)
at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:166)
at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:185)
at org.broadinstitute.hellbender.Main.instanceMain(Main.java:76)
at org.broadinstitute.hellbender.Main.main(Main.java:92)
```

Can we give a better error message when this happens?

### LINKED ISSUE #804: Need to change bounds for slice sampling of variance in copy-ratio model.
Got a weird sample (possibly with high variance?) that threw an exception.  Should be a simple fix to change the hard-coded min/max variance to instead use values derived from the data:

    java.lang.IllegalArgumentException: Initial point in slice sampler is not within specified range.
	at org.broadinstitute.hellbender.utils.Utils.validateArg(Utils.java:609)
	at org.broadinstitute.hellbender.utils.mcmc.SliceSampler.sample(SliceSampler.java:71)
	at org.broadinstitute.hellbender.tools.exome.copyratio.CopyRatioSamplers$VarianceSampler.sample(CopyRatioSamplers.java:54)
	at org.broadinstitute.hellbender.tools.exome.copyratio.CopyRatioSamplers$VarianceSampler.sample(CopyRatioSamplers.java:29)
	at org.broadinstitute.hellbender.utils.mcmc.ParameterizedModel.doGibbsUpdate(ParameterizedModel.java:134)
	at org.broadinstitute.hellbender.utils.mcmc.ParameterizedModel.update(ParameterizedModel.java:124)
	at org.broadinstitute.hellbender.utils.mcmc.GibbsSampler.runMCMC(GibbsSampler.java:76)
	at org.broadinstitute.hellbender.tools.exome.copyratio.CopyRatioModeller.fitMCMC(CopyRatioModeller.java:96)
	at org.broadinstitute.hellbender.tools.exome.ACNVModeller.fitModel(ACNVModeller.java:145)
	at org.broadinstitute.hellbender.tools.exome.ACNVModeller.performSimilarSegmentMergingIteration(ACNVModeller.java:126)
	at org.broadinstitute.hellbender.tools.exome.AllelicCNV.performSimilarSegmentMergingStep(AllelicCNV.java:391)
	at org.broadinstitute.hellbender.tools.exome.AllelicCNV.runPipeline(AllelicCNV.java:288)
	at org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram.doWork(SparkCommandLineProgram.java:38)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:108)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:166)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:185)
	at org.broadinstitute.hellbender.Main.instanceMain(Main.java:76)
	at org.broadinstitute.hellbender.Main.main(Main.java:92)

### LINKED ISSUE #827: testSubsetTargetToUsableOnes is missing dataprovider
`HDF5PCACoveragePoNCreationUtilsUnitTest.testSubsetTargetToUsableOnes()` is always skipped because its dataprovider "readCountAndPercentileData" was deleted in #634 

See here:(https://github.com/broadinstitute/gatk-protected/pull/634/files#diff-8dd9131901a75d391a8e3781d9eea0e0L465)

### LINKED ISSUE #985: Add warning/exception when target/BED files are misused
There have been a few occasions when I've gotten tripped up by GATK's expectation of a Picard-style target file (1-based, closed intervals) instead of a standard BED file (0-based, half-open intervals) for tools like `PadTargets` and `CalculateTargetCoverage`.

For instance, `PadTargets` doesn't seem to consider the half-openness of the format when adding the specified amount of padding causes two intervals to overlap:

```
> cat test.bed
contig	start	stop	name
1	100	200	chr1-a
1	250	350	chr1-b

> java -jar /gatk-protected.jar PadTargets -T test.bed -O test.padded.bed --padding 75

> cat test.padded.bed
contig	start	stop	name
1	25	225	chr1-a
1	226	425	chr1-b
```

This *does not* contain the 0-based position 225 (a.k.a. chr1:226 if you're using IGV). If you arbitrarily add that position back to one of the intervals (i.e. `1 25 226`), `CalculateTargetCoverage` will raise an exception about overlapping intervals, despite this being a completely valid BED file.

Granted, both tools do ask for a "target" file, but it's not clear that they mean a Picard-style list of intervals. I suspect many people are using BED files instead, including our very own @LeeTL1220 in the [CNV workflow instructions](http://gatkforums.broadinstitute.org/gatk/discussion/6791/description-and-examples-of-the-steps-in-the-cnv-case-and-cnv-pon-creation-workflows#create-pon-workflow).

Maybe he read [this FAQ](http://gatkforums.broadinstitute.org/gatk/discussion/1319/collected-faqs-about-interval-lists), which claims that the GATK should automatically convert these intervals as needed. So, there's apparently some confusion about whether any of this actually _should_ work.

I don't mind converting the BED file because it's easy and 0-based indexing is stupid (although `ConvertBedToTargetFile` doesn't seem to work—I'll submit a separate issue), but I think either the BED file should be converted internally or a warning/exception should be raised if that behavior is too opinionated.

### LINKED ISSUE #986: ConvertBedToTargetFile raises codec exception
```
> cat test2.bed
contig	start	stop	name
1	100	200	chr1-a
1	250	350	chr1-b
> java -jar gatk-protected.jar ConvertBedToTargetFile -I test.bed -O test.target
```
results in an exception:

>Multiple codecs found able to decode file... This indicates a misconfiguration on the part of the codec authors. Matching codecs are: htsjdk.tribble.bed.BEDCodec org.broadinstitute.hellbender.utils.codecs.TargetCodec

[Full log is here](https://gist.github.com/dpmccabe/8b4208c1cc98998fbc508592426364f9). I get this error both on RedHat and OS X.

### PR DESCRIPTION
**Pull request by samuelklee**
-Removed repeat fraction from AnnotateTargets.
Closes #645.

-Fixed BEDCodec and TargetCodec conflict in ConvertBedToTargetFile.
Closes #985, closes #986.

-Moved DataProvider for testSubsetTargetToUsableOnes.
Closes #827.

-Added exception for SNPSegmenter.
Closes #684.

-Removed hard-coded upper bound for copy-ratio variance slice sampling.
Closes #804.

-Removed some scripts for test data generation.

### CODE REVIEW

**Comment by samuelklee** [comment]
Some minor fixes.  Please review, @asmirnov239.

File: src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioModeller.java [hunk 1]
```diff
@@ -50,10 +49,11 @@ public CopyRatioModeller(final SegmentedGenome segmentedGenome) {
         //set widths for slice sampling of variance and segment-mean posteriors using empirical variance estimate.
         //variance posterior is inverse chi-squared, segment-mean posteriors are Gaussian; the below expressions
         //approximate the standard deviations of these distributions.
-        final double varianceEstimate = data.estimateVariance();
-        final double varianceSliceSamplingWidth = Math.sqrt(2. * varianceEstimate / data.getNumTargets()) / 10.;    //take the width down an order of magnitude to account for inflation by outliers
         final double coverageMin = data.getCoverageMin();
         final double coverageMax = data.getCoverageMax();
+        final double varianceEstimate = data.estimateVariance();
+        final double varianceSliceSamplingWidth = Math.sqrt(2. * varianceEstimate / data.getNumTargets()) / 10.;    //take the width down an order of magnitude to account for inflation by outliers
+        final double varianceMax = Math.abs(coverageMax - coverageMin);
```
Comment (asmirnov239) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioModeller.java]: Couldn't the variance be proportional to the square of this value in the worst case?

Comment (asmirnov239) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioModeller.java]: Also you could add a unit test for the edge case with couple of targets with different coverage

Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioModeller.java]: In principle, it could be even worse.  If I only see a single point near zero, there's still some probability that it was drawn from a distribution with large variance---it's just less likely the larger the variance.

I think this is a reasonable range, but might add some additional checks and tests.

Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioModeller.java]: As we discussed, I'm not sure if there's a sensible unit test here.  Merging!
